docs/source/references/apollo-config.md

```diff
@@ -0,0 +1,201 @@
+
+If you have an Engine service set up and tracking your schema's history, you can point your client to that very simply by putting the id of your service in your Apollo config:
+
+```js line=3
+module.exports = {
+  client: {
+    service: 'my-apollo-service', // the id of your service in Engine (from the URL)
+  },
+};
+```
+
+If you're tracking different versions of your schema in the registry using schema tags, you can include the tag you're interested in linking your client to like so:
+
+```js line=3
+module.exports = {
+  client: {
+    service: 'my-apollo-service@beta', // "beta" is the schema tag we're using
+  },
+};
+```
+If a schema tag is not specified, the default value is `current`.
+
+#### Option 2: Link a schema from a remote endpoint
+
+Remote endpoints can be used to pull down a schema from a running service. This can be configured like so:
+
+```js line=3-11
+module.exports = {
+  client: {
+    service: {
+      name: 'github',
+      url: 'https://api.github.com/graphql',
+      // optional headers
+      headers: {
+        authorization: 'Bearer lkjfalkfjadkfjeopknavadf',
+      },
+      // optional disable SSL validation check
+      skipSSLValidation: true,
+    },
+  },
+};
+```
+
+#### Option 3: Link a schema from a local file
+
+In some cases, teams may have a local generated file of their schema they want to link. This can be either a `.graphql` file with the schma in SDL, or a saved introspection result in `.json`. To link a client with a local file, configure the project like this:
+
+```js line=3-6
+module.exports = {
+  client: {
+    service: {
+      name: 'my-service-name',
+      localSchemaFile: './path/to/schema.graphql',
+    },
+  },
+};
+```
+
+### `includes`, `excludes`
+
+*Optional* –– by default, Apollo tools will look under a `./src` directory to find all operations and SDL to extract.
+
+Client projects often contain client-side schema definitions for local state with Apollo Client. To make sure the Apollo CLI and VS Code extension can find these files and read them correctly, you may need to add some additional definitions in your config file. To configure which folders to include, and which to ignore, adjust your config like so:
+
+```js line=3-4
+module.exports = {
+  client: {
+    includes: ['./imports/**/*.js'],
+    excludes: ['**/__tests__/**/*'],
+  },
+};
+```
+
+### `tagName`
+
+*Optional* –– custom tagged template literal.
+
+When using GraphQL with JavaScript or TypeScript projects, it is common to use the `gql` tagged template literal to write out operations. Apollo tools will be looking through your files for the `gql` tag to extract your queries, so if you use a different template literal, you can configure it like so:
+
+```js line=3
+module.exports = {
+  client: {
+    tagName: 'graphql',
+  },
+};
+```
+
+### `addTypename`
+
+*Optional* –– Apollo will by default add the `__typename` field to all operations automatically.
```

Comment:
We should recommend that this option be 1:1 with how their `ApolloClient` is configured. We might also say directly that this will add `__typename` to all the generated types during codegen, since right now that's all this field is used for.

Reply:
Maybe we should have Apollo Client read that preference from the config directly if it exists?